synapse/storage/databases/main/events_worker.py

```diff
@@ -588,12 +590,14 @@ async def get_events_as_list(
         return events
 
     @cancellable
-    async def get_events_from_cache_or_db(
+    async def get_unredacted_events_from_cache_or_db(
         self,
         event_ids: Iterable[str],
         allow_rejected: bool = False,
     ) -> Dict[str, EventCacheEntry]:
-        """Fetch a bunch of events from the cache or the database.
+        """Fetch a bunch of events from the cache or the database. Please note that the
+        events pulled by this function will not have any redactions applied, and no guarantee
+        is made about the ordering of the events returned.
```

Comment:
Convention for docstrings is to start with a one-line summary, then add more context in separate paragraphs:

```suggestion
        """Fetch a bunch of events from the cache or the database. 
        
        Note that the events pulled by this function will not have any redactions applied, and
        no guarantee is made about the ordering of the events returned.
```